torch/mps/__init__.py

```diff
@@ -0,0 +1,88 @@
+r"""
+This unit enables an interface for accessing MPS backend in python
+"""
+import torch
+from functools import lru_cache
+import threading
+from .. import Tensor
+import torch._C
+
+_initialized = False
+_initialization_lock = threading.Lock()
+default_mps_generator: torch._C.Generator = None  # type: ignore[assignment]
+
+def init():
+    r"""Initialize PyTorch's MPS state.
+    Does nothing if the MPS state is already initialized.
+    """
+    _lazy_init()
+
+def _lazy_init():
+    global _initialized
+    if is_initialized():
+        return
+    with _initialization_lock:
+        if is_initialized():
+            return
+        if not hasattr(torch._C, '_mps_init'):
+            raise AssertionError("Torch not compiled with MPS enabled")
+        torch._C._mps_init()
+        _initialized = True
+
+@lru_cache()
+def is_available() -> bool:
+    r"""Returns a bool indicating if MPS is currently available."""
+    if not hasattr(torch._C, '_is_mps_available'):
+        return False
+    return torch._C._is_mps_available()
+
+@lru_cache()
+def is_macos13_or_newer() -> bool:
+    r"""Returns a bool indicating whether MPS is running on MacOS 13 or newer."""
+    if not hasattr(torch._C, '_is_mps_on_macos_13_or_newer'):
+        return False
+    return torch._C._is_mps_on_macos_13_or_newer()
```

Comment:
Why this needs to be part of `torch.mps` package? Isn't it an implementation detail that usually specific to a backend?

Reply:
Updated the PR.